<a href="https://colab.research.google.com/github/Sedowlas/Cat-Boost/blob/main/Spaceship_titanic(EDA%2C_ML%2C_PD%2C_SQL).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

PassengerId - Уникальный идентификатор для каждого пассажира. Каждый идентификатор принимает форму gggg_pp, в которой gggg указывается группа, с которой путешествует пассажир, и pp их номер в группе. Люди в группе часто являются членами семьи, но не всегда.  
HomePlanet - Планета, с которой вылетел пассажир, обычно планета их постоянного проживания.  
CryoSleep - Указывает, решил ли пассажир быть помещенным в анабиоз на время полета. Пассажиры в криосне прикованы к своим каютам.  
Cabin - Номер каюты, в которой находится пассажир. Принимает формуdeck/num/side, где sideможет быть либо Pпо левому, либо Sпо правому борту.  
Destination - Планета, на которую высадится пассажир.  
Age - Возраст пассажира.  
VIP - Оплатил ли пассажир специальное VIP-обслуживание во время рейса.  
RoomService, FoodCourt, ShoppingMall, Spa, VRDeck- Сумма, которую пассажир выставил за каждый из множества роскошных удобств космического корабля Титаник.  
Name - Имя и фамилия пассажира.  
Transported - Был ли пассажир перенесен в другое измерение. Это цель, столбец, который вы пытаетесь предсказать.  

#Загрузка данных

https://drive.google.com/file/d/13A1JWS9sh70LO3ZOLco4-XWvRwc6yT4z/view?usp=sharing

In [1]:
!gdown 13A1JWS9sh70LO3ZOLco4-XWvRwc6yT4z

Downloading...
From: https://drive.google.com/uc?id=13A1JWS9sh70LO3ZOLco4-XWvRwc6yT4z
To: /content/spaceship-titanic.zip
100% 306k/306k [00:00<00:00, 110MB/s]


In [2]:
!unzip /content/spaceship-titanic.zip

Archive:  /content/spaceship-titanic.zip
replace sample_submission.csv? [y]es, [n]o, [A]ll, [N]one, [r]ename: 

#Предобработка

In [3]:
import pandas as pd
import numpy as np

In [4]:
import sqlite3

In [5]:
df = pd.read_csv('/content/train.csv')

In [6]:
con = sqlite3.connect('db')

In [7]:
df.head()

PassengerId HomePlanet CryoSleep  Cabin  Destination   Age    VIP  \
0     0001_01     Europa     False  B/0/P  TRAPPIST-1e  39.0  False   
1     0002_01      Earth     False  F/0/S  TRAPPIST-1e  24.0  False   
2     0003_01     Europa     False  A/0/S  TRAPPIST-1e  58.0   True   
3     0003_02     Europa     False  A/0/S  TRAPPIST-1e  33.0  False   
4     0004_01      Earth     False  F/1/S  TRAPPIST-1e  16.0  False   

   RoomService  FoodCourt  ShoppingMall     Spa  VRDeck               Name  \
0          0.0        0.0           0.0     0.0     0.0    Maham Ofracculy   
1        109.0        9.0          25.0   549.0    44.0       Juanna Vines   
2         43.0     3576.0           0.0  6715.0    49.0      Altark Susent   
3          0.0     1283.0         371.0  3329.0   193.0       Solam Susent   
4        303.0       70.0         151.0   565.0     2.0  Willy Santantines   

   Transported  
0        False  
1         True  
2        False  
3        False  
4         True

In [8]:
df['Transported'] = (-df['Transported']).astype(np.int8)

In [9]:
len(df)

8693

In [10]:
df.columns = df.columns.str.lower()

In [11]:
df.columns = ['passenger_id', 'home_planet', 'cryo_sleep', 'cabin', 'destination', 'age',
       'vip', 'room_service', 'food_court', 'shopping_mall', 'spa', 'vr_deck',
       'name', 'transported']

In [12]:
df.isna().mean()

passenger_id     0.000000
home_planet      0.023122
cryo_sleep       0.024963
cabin            0.022892
destination      0.020936
age              0.020591
vip              0.023352
room_service     0.020821
food_court       0.021051
shopping_mall    0.023927
spa              0.021051
vr_deck          0.021627
name             0.023007
transported      0.000000
dtype: float64

In [13]:
df[['home_planet','cryo_sleep','cabin','destination','vip','name']] = df[['home_planet','cryo_sleep','cabin','destination','vip','name']].fillna('no_info')

In [14]:
df[['age','room_service','food_court','shopping_mall','spa','vr_deck']] = df[['age','room_service','food_court','shopping_mall','spa','vr_deck']].fillna(0)

In [15]:
df.dtypes

passenger_id      object
home_planet       object
cryo_sleep        object
cabin             object
destination       object
age              float64
vip               object
room_service     float64
food_court       float64
shopping_mall    float64
spa              float64
vr_deck          float64
name              object
transported         int8
dtype: object

In [16]:
df.to_sql('space',con,index=False,if_exists='replace')

In [17]:
def select(sql):
  return pd.read_sql(sql,con)

#Анализ

In [18]:
df.head()

passenger_id home_planet cryo_sleep  cabin  destination   age    vip  \
0      0001_01      Europa      False  B/0/P  TRAPPIST-1e  39.0  False   
1      0002_01       Earth      False  F/0/S  TRAPPIST-1e  24.0  False   
2      0003_01      Europa      False  A/0/S  TRAPPIST-1e  58.0   True   
3      0003_02      Europa      False  A/0/S  TRAPPIST-1e  33.0  False   
4      0004_01       Earth      False  F/1/S  TRAPPIST-1e  16.0  False   

   room_service  food_court  shopping_mall     spa  vr_deck  \
0           0.0         0.0            0.0     0.0      0.0   
1         109.0         9.0           25.0   549.0     44.0   
2          43.0      3576.0            0.0  6715.0     49.0   
3           0.0      1283.0          371.0  3329.0    193.0   
4         303.0        70.0          151.0   565.0      2.0   

                name  transported  
0    Maham Ofracculy            1  
1       Juanna Vines            0  
2      Altark Susent            1  
3       Solam Susent            1  
4  Willy Santantines            0

In [19]:
df.groupby(['home_planet'])[['age']].agg(['count','mean'])

age           
            count       mean
home_planet                 
Earth        4602  25.569752
Europa       2131  33.676678
Mars         1759  28.581012
no_info       201  28.507463

In [20]:
sql = '''select t.passenger_id, t.name, t.room_service from space t
order by t.room_service desc
'''

In [21]:
select(sql)

passenger_id                name  room_service
0         4690_02     Tope Dishocatal       14327.0
1         5455_01         Varkes Nate        9920.0
2         9202_01        Alark Eguing        8586.0
3         8470_01   Brafirk Aloubtled        8243.0
4         7585_02  Azhasim Admingried        8209.0
...           ...                 ...           ...
8688      9275_03     Atlasym Conable           0.0
8689      9276_01   Gravior Noxnuther           0.0
8690      9278_01     Kurta Mondalley           0.0
8691      9279_01        Fayey Connon           0.0
8692      9280_01    Celeon Hontichre           0.0

[8693 rows x 3 columns]

Самый большой чек за комнату составил 14327

In [22]:
sql = '''select t.passenger_id, t.name, t.food_court from space t
order by t.food_court desc
'''

In [23]:
select(sql)

passenger_id                name  food_court
0         2215_01  Pulchib Quidedbolt     29813.0
1         1292_01      Scharab Conale     27723.0
2         3444_01    Ainoxa Scerodbox     27071.0
3         3796_01     Dhena Semageary     26830.0
4         1969_01      Unukath Sandul     21066.0
...           ...                 ...         ...
8688      9270_01       Lan Mckinsond         0.0
8689      9274_01     Chelsa Bullisey         0.0
8690      9275_01     Polaton Conable         0.0
8691      9278_01     Kurta Mondalley         0.0
8692      9279_01        Fayey Connon         0.0

[8693 rows x 3 columns]

Самый большой чек за фуд корт составил 29813

In [24]:
sql = '''select t.passenger_id, t.name, t.shopping_mall from space t
order by t.shopping_mall desc
'''

In [25]:
select(sql)

passenger_id                name  shopping_mall
0         8989_01             no_info        23492.0
1         6583_01      Charda Sunlove        12253.0
2         7941_01  Alrakan Apedishaft        10705.0
3         5040_01      Enibik Dynigic        10424.0
4         6027_01   Graviph Coweboded         9058.0
...           ...                 ...            ...
8688      9275_03     Atlasym Conable            0.0
8689      9276_01   Gravior Noxnuther            0.0
8690      9278_01     Kurta Mondalley            0.0
8691      9280_01    Celeon Hontichre            0.0
8692      9280_02    Propsh Hontichre            0.0

[8693 rows x 3 columns]

Самый большой чек за торговый центр составил 23492

In [26]:
sql = '''select t.passenger_id, t.name, t.spa from space t
order by t.spa desc
'''

In [27]:
select(sql)

passenger_id               name      spa
0         6064_01  Markar Radisiouss  22408.0
1         8556_01   Mirfar Optionful  18572.0
2         1455_05     Antino Wassird  16594.0
3         1161_02  Mirfark Anderking  16139.0
4         7339_01    Ainkard Fanseld  15586.0
...           ...                ...      ...
8688      9272_01      Ireene Simson      0.0
8689      9274_01    Chelsa Bullisey      0.0
8690      9275_01    Polaton Conable      0.0
8691      9278_01    Kurta Mondalley      0.0
8692      9280_02   Propsh Hontichre      0.0

[8693 rows x 3 columns]

Самый дорогой чек за спа составил 22408

In [28]:
sql = '''select t.passenger_id, t.name, t.vr_deck from space t
order by t.vr_deck desc
'''

In [29]:
select(sql)

passenger_id             name  vr_deck
0         5977_02          no_info  24133.0
1         4594_02   Tabiton Folhal  20336.0
2         0768_01   Maiam Oilloody  17306.0
3         6908_01  Sheleon Swinvul  17074.0
4         3619_01    Tauva Roureed  16337.0
...           ...              ...      ...
8688      9272_01    Ireene Simson      0.0
8689      9274_01  Chelsa Bullisey      0.0
8690      9275_01  Polaton Conable      0.0
8691      9278_01  Kurta Mondalley      0.0
8692      9279_01     Fayey Connon      0.0

[8693 rows x 3 columns]

Самый дорогой чек за VR палубу составил 24133

In [30]:
sql = '''select t.vip from space t 
where t.vip = '0'
'''

In [31]:
select(sql)

vip
0      0
1      0
2      0
3      0
4      0
...   ..
8286   0
8287   0
8288   0
8289   0
8290   0

[8291 rows x 1 columns]

Узнали сколько человек не оплатили VIP места - 8291  
False - 0

In [32]:
sql = '''select t.vip from space t 
where t.vip = '1'
'''

In [33]:
select(sql)

vip
0     1
1     1
2     1
3     1
4     1
..   ..
194   1
195   1
196   1
197   1
198   1

[199 rows x 1 columns]

Узнали что 199 пассажиров оплатили VIP места  
Then - 1

In [34]:
sql = '''select * from space t 
where t.vip = '1'
'''

In [35]:
select(sql)

passenger_id home_planet cryo_sleep    cabin  destination   age vip  \
0        0003_01      Europa          0    A/0/S  TRAPPIST-1e  58.0   1   
1        0112_01      Europa          0    B/1/S  55 Cancri e  48.0   1   
2        0128_01        Mars          0    D/3/S  TRAPPIST-1e  61.0   1   
3        0224_01        Mars          0   F/42/S  TRAPPIST-1e  32.0   1   
4        0321_01     no_info          0   F/61/S  TRAPPIST-1e  59.0   1   
..           ...         ...        ...      ...          ...   ...  ..   
194      9158_01      Europa          1  B/298/P  55 Cancri e  30.0   1   
195      9194_02      Europa          0  E/603/S  TRAPPIST-1e  32.0   1   
196      9197_02      Europa          0  C/308/P      no_info  41.0   1   
197      9230_01      Europa          0  C/342/S  TRAPPIST-1e  36.0   1   
198      9276_01      Europa          0   A/98/P  55 Cancri e  41.0   1   

     room_service  food_court  shopping_mall     spa  vr_deck  \
0            43.0      3576.0            0.0  6715.0     49.0   
1             0.0      2537.0           87.0    17.0     13.0   
2          2353.0       334.0            9.0   316.0      2.0   
3           181.0         0.0            5.0  1634.0      0.0   
4          1018.0         0.0          209.0     0.0      0.0   
..            ...         ...            ...     ...      ...   
194           0.0         0.0            0.0     0.0      0.0   
195        1003.0       909.0            0.0     0.0     15.0   
196           0.0      7964.0            0.0  3238.0   5839.0   
197           0.0      5600.0          715.0  2868.0    971.0   
198           0.0      6819.0            0.0  1643.0     74.0   

                  name  transported  
0        Altark Susent            1  
1         Moth Cowtale            0  
2           Grohs Fles            1  
3          Blues Queen            1  
4         Quites Bache            1  
..                 ...          ...  
194    Magnon Maglible            0  
195      Tachba Subwor            1  
196     Aludram Platch            1  
197            no_info            0  
198  Gravior Noxnuther            1  

[199 rows x 14 columns]

Узнали всю информации о тех пассажирах которые приобрели VIP места

In [36]:
df['destination'].value_counts()

TRAPPIST-1e      5915
55 Cancri e      1800
PSO J318.5-22     796
no_info           182
Name: destination, dtype: int64

Узнали какие планеты посещает космический титаник и на каких планетах больше всего выходят пассажиры

In [37]:
sql = '''select t.destination, t.transported from space t 
where t.destination = 'TRAPPIST-1e' and t.transported = '0'
'''

In [38]:
select(sql)

destination  transported
0     TRAPPIST-1e            0
1     TRAPPIST-1e            0
2     TRAPPIST-1e            0
3     TRAPPIST-1e            0
4     TRAPPIST-1e            0
...           ...          ...
2782  TRAPPIST-1e            0
2783  TRAPPIST-1e            0
2784  TRAPPIST-1e            0
2785  TRAPPIST-1e            0
2786  TRAPPIST-1e            0

[2787 rows x 2 columns]

Узнали сколько человек не добрались до планеты TRAPPIST-1e - 3128

In [39]:
sql = '''select t.destination, t.transported from space t 
where t.destination = 'TRAPPIST-1e' and t.transported = '1'
'''

In [40]:
select(sql)

destination  transported
0     TRAPPIST-1e            1
1     TRAPPIST-1e            1
2     TRAPPIST-1e            1
3     TRAPPIST-1e            1
4     TRAPPIST-1e            1
...           ...          ...
3123  TRAPPIST-1e            1
3124  TRAPPIST-1e            1
3125  TRAPPIST-1e            1
3126  TRAPPIST-1e            1
3127  TRAPPIST-1e            1

[3128 rows x 2 columns]

Узнали сколько человек добрались до планеты TRAPPIST-1e - 2787

In [41]:
sql = '''select t.destination, t.transported from space t 
where t.destination = '55 Cancri e' and t.transported = '0'
'''

In [42]:
select(sql)

destination  transported
0     55 Cancri e            0
1     55 Cancri e            0
2     55 Cancri e            0
3     55 Cancri e            0
4     55 Cancri e            0
...           ...          ...
1093  55 Cancri e            0
1094  55 Cancri e            0
1095  55 Cancri e            0
1096  55 Cancri e            0
1097  55 Cancri e            0

[1098 rows x 2 columns]

Узнали сколько человек не добрались до планеты 55 Cancri e - 702

In [43]:
sql = '''select t.destination, t.transported from space t 
where t.destination = '55 Cancri e' and t.transported = '1'
'''

In [44]:
select(sql)

destination  transported
0    55 Cancri e            1
1    55 Cancri e            1
2    55 Cancri e            1
3    55 Cancri e            1
4    55 Cancri e            1
..           ...          ...
697  55 Cancri e            1
698  55 Cancri e            1
699  55 Cancri e            1
700  55 Cancri e            1
701  55 Cancri e            1

[702 rows x 2 columns]

Узнали сколько человек добрались до планеты 55 Cancri e - 1098

In [45]:
sql = '''select t.destination, t.transported from space t 
where t.destination = 'PSO J318.5-22' and t.transported = '0'
'''

In [46]:
select(sql)

destination  transported
0    PSO J318.5-22            0
1    PSO J318.5-22            0
2    PSO J318.5-22            0
3    PSO J318.5-22            0
4    PSO J318.5-22            0
..             ...          ...
396  PSO J318.5-22            0
397  PSO J318.5-22            0
398  PSO J318.5-22            0
399  PSO J318.5-22            0
400  PSO J318.5-22            0

[401 rows x 2 columns]

Узнали сколько человек не добрались до планеты PSO J318.5-22 - 395

In [47]:
sql = '''select t.destination, t.transported from space t 
where t.destination = 'PSO J318.5-22' and t.transported = '1'
'''

In [48]:
select(sql)

destination  transported
0    PSO J318.5-22            1
1    PSO J318.5-22            1
2    PSO J318.5-22            1
3    PSO J318.5-22            1
4    PSO J318.5-22            1
..             ...          ...
390  PSO J318.5-22            1
391  PSO J318.5-22            1
392  PSO J318.5-22            1
393  PSO J318.5-22            1
394  PSO J318.5-22            1

[395 rows x 2 columns]

Узнали сколько человек добрались до планеты PSO J318.5-22 - 401

#Разбивка данных на train, val и test

In [49]:
from sklearn.model_selection import train_test_split

In [50]:
train, test = train_test_split(df,train_size=0.6,random_state=42)

In [51]:
val, test = train_test_split(test,train_size=0.5,random_state=42)

In [52]:
train['transported'].mean()

0.4987535953978907

In [53]:
val['transported'].mean()

0.48821161587119033

In [54]:
test['transported'].mean()

0.4974123059229442

#Машинное обучение

In [55]:
! pip install catboost

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [56]:
from catboost import CatBoostClassifier

In [57]:
train_full = pd.concat([train,val])

In [58]:
train_full.columns

Index(['passenger_id', 'home_planet', 'cryo_sleep', 'cabin', 'destination',
       'age', 'vip', 'room_service', 'food_court', 'shopping_mall', 'spa',
       'vr_deck', 'name', 'transported'],
      dtype='object')

In [59]:
X = ['passenger_id', 'home_planet', 'cryo_sleep', 'cabin', 'destination',
       'age', 'vip', 'room_service', 'food_court', 'shopping_mall', 'spa',
       'vr_deck', 'name', 'transported']

cat_features = ['passenger_id', 'home_planet', 'cryo_sleep', 'cabin', 'destination', 'vip', 'name']

y = ['transported']

In [60]:
params = {'cat_features' : cat_features,
            'verbose':100,
          'random_seed':42,}

In [61]:
model = CatBoostClassifier(**params)

In [62]:
model.fit(train[X],train[y],eval_set=(val[X],val[y]))

Learning rate set to 0.047651
0:	learn: 0.5373437	test: 0.5376521	best: 0.5376521 (0)	total: 75.4ms	remaining: 1m 15s
100:	learn: 0.0004057	test: 0.0003226	best: 0.0003226 (100)	total: 1.34s	remaining: 12s
200:	learn: 0.0003488	test: 0.0002713	best: 0.0002713 (200)	total: 2.48s	remaining: 9.86s
300:	learn: 0.0002908	test: 0.0002183	best: 0.0002183 (300)	total: 3.49s	remaining: 8.1s
400:	learn: 0.0002904	test: 0.0002182	best: 0.0002182 (400)	total: 4.41s	remaining: 6.58s
500:	learn: 0.0002661	test: 0.0001971	best: 0.0001971 (500)	total: 5.32s	remaining: 5.3s
600:	learn: 0.0002657	test: 0.0001971	best: 0.0001971 (599)	total: 6.25s	remaining: 4.15s
700:	learn: 0.0002650	test: 0.0001969	best: 0.0001969 (700)	total: 7.16s	remaining: 3.06s
800:	learn: 0.0002645	test: 0.0001967	best: 0.0001967 (800)	total: 8.08s	remaining: 2.01s
900:	learn: 0.0002633	test: 0.0001966	best: 0.0001966 (900)	total: 9.05s	remaining: 994ms
999:	learn: 0.0002565	test: 0.0001965	best: 0.0001965 (999)	total: 10s	remai

In [63]:
test['pred_all'] = model.predict(test[X])

In [64]:
test.head()

passenger_id home_planet cryo_sleep     cabin    destination   age  \
1209      1288_01       Earth      False   G/192/P    TRAPPIST-1e  24.0   
4983      5316_01        Mars       True  F/1087/P    55 Cancri e  23.0   
7087      7545_01       Earth       True  G/1225/S    TRAPPIST-1e   0.0   
308       0341_01       Earth       True    G/57/P  PSO J318.5-22   5.0   
7740      8260_03       Earth       True  G/1334/S    55 Cancri e   4.0   

        vip  room_service  food_court  shopping_mall  spa  vr_deck  \
1209  False         621.0         0.0            0.0  0.0      0.0   
4983  False           0.0         0.0            0.0  0.0      0.0   
7087  False           0.0         0.0            0.0  0.0      0.0   
308   False           0.0         0.0            0.0  0.0      0.0   
7740  False           0.0         0.0            0.0  0.0      0.0   

                    name  transported  pred_all  
1209      Nance Flemaney            1         1  
4983            Cats Ses            0         0  
7087  Jasony Nguerrettye            1         1  
308   Lorian Webstephrey            1         1  
7740    Melara Daviddley            0         0

#Test

In [65]:
dff = pd.read_csv('/content/test.csv')

In [66]:
dff.head()

PassengerId HomePlanet CryoSleep  Cabin  Destination   Age    VIP  \
0     0013_01      Earth      True  G/3/S  TRAPPIST-1e  27.0  False   
1     0018_01      Earth     False  F/4/S  TRAPPIST-1e  19.0  False   
2     0019_01     Europa      True  C/0/S  55 Cancri e  31.0  False   
3     0021_01     Europa     False  C/1/S  TRAPPIST-1e  38.0  False   
4     0023_01      Earth     False  F/5/S  TRAPPIST-1e  20.0  False   

   RoomService  FoodCourt  ShoppingMall     Spa  VRDeck              Name  
0          0.0        0.0           0.0     0.0     0.0   Nelly Carsoning  
1          0.0        9.0           0.0  2823.0     0.0    Lerome Peckers  
2          0.0        0.0           0.0     0.0     0.0   Sabih Unhearfus  
3          0.0     6652.0           0.0   181.0   585.0  Meratz Caltilter  
4         10.0        0.0         635.0     0.0     0.0   Brence Harperez

In [67]:
dff.columns = dff.columns.str.lower()

In [68]:
dff.columns = ['passenger_id', 'home_planet', 'cryo_sleep', 'cabin', 'destination', 'age',
       'vip', 'room_service', 'food_court', 'shopping_mall', 'spa', 'vr_deck',
       'name']

In [69]:
dff.isna().mean()

passenger_id     0.000000
home_planet      0.020341
cryo_sleep       0.021744
cabin            0.023381
destination      0.021510
age              0.021277
vip              0.021744
room_service     0.019172
food_court       0.024784
shopping_mall    0.022913
spa              0.023615
vr_deck          0.018705
name             0.021978
dtype: float64

In [70]:
dff[['home_planet','cryo_sleep','cabin','destination','vip','name']] = df[['home_planet','cryo_sleep','cabin','destination','vip','name']].fillna('no_info')

In [71]:
dff[['age','room_service','food_court','shopping_mall','spa','vr_deck']] = df[['age','room_service','food_court','shopping_mall','spa','vr_deck']].fillna(0)

In [72]:
X = ['passenger_id', 'home_planet', 'cryo_sleep', 'cabin', 'destination',
       'age', 'vip', 'room_service', 'food_court', 'shopping_mall', 'spa',
       'vr_deck', 'name', 'transported']

cat_features = ['passenger_id', 'home_planet', 'cryo_sleep', 'cabin', 'destination', 'vip', 'name']

y = ['transported']

In [73]:
params = {'cat_features' : cat_features,
            'verbose':100,
          'random_seed':42,}

In [74]:
model = CatBoostClassifier(**params)

In [75]:
model.fit(train[X],train[y],eval_set=(val[X],val[y]))

Learning rate set to 0.047651
0:	learn: 0.5373437	test: 0.5376521	best: 0.5376521 (0)	total: 18.1ms	remaining: 18.1s
100:	learn: 0.0004057	test: 0.0003226	best: 0.0003226 (100)	total: 1.32s	remaining: 11.8s
200:	learn: 0.0003488	test: 0.0002713	best: 0.0002713 (200)	total: 2.43s	remaining: 9.65s
300:	learn: 0.0002908	test: 0.0002183	best: 0.0002183 (300)	total: 3.47s	remaining: 8.07s
400:	learn: 0.0002904	test: 0.0002182	best: 0.0002182 (400)	total: 4.41s	remaining: 6.59s
500:	learn: 0.0002661	test: 0.0001971	best: 0.0001971 (500)	total: 5.35s	remaining: 5.33s
600:	learn: 0.0002657	test: 0.0001971	best: 0.0001971 (599)	total: 6.27s	remaining: 4.16s
700:	learn: 0.0002650	test: 0.0001969	best: 0.0001969 (700)	total: 7.19s	remaining: 3.07s
800:	learn: 0.0002645	test: 0.0001967	best: 0.0001967 (800)	total: 8.11s	remaining: 2.02s
900:	learn: 0.0002633	test: 0.0001966	best: 0.0001966 (900)	total: 9.06s	remaining: 995ms
999:	learn: 0.0002565	test: 0.0001965	best: 0.0001965 (999)	total: 9.98s	

In [76]:
dff.head()

passenger_id home_planet cryo_sleep  cabin  destination   age    vip  \
0      0013_01      Europa      False  B/0/P  TRAPPIST-1e  39.0  False   
1      0018_01       Earth      False  F/0/S  TRAPPIST-1e  24.0  False   
2      0019_01      Europa      False  A/0/S  TRAPPIST-1e  58.0   True   
3      0021_01      Europa      False  A/0/S  TRAPPIST-1e  33.0  False   
4      0023_01       Earth      False  F/1/S  TRAPPIST-1e  16.0  False   

   room_service  food_court  shopping_mall     spa  vr_deck               name  
0           0.0         0.0            0.0     0.0      0.0    Maham Ofracculy  
1         109.0         9.0           25.0   549.0     44.0       Juanna Vines  
2          43.0      3576.0            0.0  6715.0     49.0      Altark Susent  
3           0.0      1283.0          371.0  3329.0    193.0       Solam Susent  
4         303.0        70.0          151.0   565.0      2.0  Willy Santantines

In [77]:
test['space'] = model.predict_proba(test[X])[:,1]

In [78]:
test['space'] = model.predict_proba(test[X])[:,1]

In [79]:
test.head()

passenger_id home_planet cryo_sleep     cabin    destination   age  \
1209      1288_01       Earth      False   G/192/P    TRAPPIST-1e  24.0   
4983      5316_01        Mars       True  F/1087/P    55 Cancri e  23.0   
7087      7545_01       Earth       True  G/1225/S    TRAPPIST-1e   0.0   
308       0341_01       Earth       True    G/57/P  PSO J318.5-22   5.0   
7740      8260_03       Earth       True  G/1334/S    55 Cancri e   4.0   

        vip  room_service  food_court  shopping_mall  spa  vr_deck  \
1209  False         621.0         0.0            0.0  0.0      0.0   
4983  False           0.0         0.0            0.0  0.0      0.0   
7087  False           0.0         0.0            0.0  0.0      0.0   
308   False           0.0         0.0            0.0  0.0      0.0   
7740  False           0.0         0.0            0.0  0.0      0.0   

                    name  transported  pred_all     space  
1209      Nance Flemaney            1         1  0.999878  
4983            Cats Ses            0         0  0.000150  
7087  Jasony Nguerrettye            1         1  0.999618  
308   Lorian Webstephrey            1         1  0.999599  
7740    Melara Daviddley            0         0  0.000161

In [80]:
test['space_score'] = (test['space'] > 0.5) * 1

In [81]:
test['space_score'] = (test['space'] > 0.5) * 1

In [82]:
test.head()

passenger_id home_planet cryo_sleep     cabin    destination   age  \
1209      1288_01       Earth      False   G/192/P    TRAPPIST-1e  24.0   
4983      5316_01        Mars       True  F/1087/P    55 Cancri e  23.0   
7087      7545_01       Earth       True  G/1225/S    TRAPPIST-1e   0.0   
308       0341_01       Earth       True    G/57/P  PSO J318.5-22   5.0   
7740      8260_03       Earth       True  G/1334/S    55 Cancri e   4.0   

        vip  room_service  food_court  shopping_mall  spa  vr_deck  \
1209  False         621.0         0.0            0.0  0.0      0.0   
4983  False           0.0         0.0            0.0  0.0      0.0   
7087  False           0.0         0.0            0.0  0.0      0.0   
308   False           0.0         0.0            0.0  0.0      0.0   
7740  False           0.0         0.0            0.0  0.0      0.0   

                    name  transported  pred_all     space  space_score  
1209      Nance Flemaney            1         1  0.999878            1  
4983            Cats Ses            0         0  0.000150            0  
7087  Jasony Nguerrettye            1         1  0.999618            1  
308   Lorian Webstephrey            1         1  0.999599            1  
7740    Melara Daviddley            0         0  0.000161            0

In [83]:
test.columns = ['PassengerId', 'home_planet', 'cryo_sleep', 'cabin', 'destination', 'age', 'vip',
       'room_service', 'food_court', 'shopping_mall', 'spa', 'vr_deck', 'name', 'transported', 'pred_all', 'space', 'Transported']

In [84]:
df_new =test[['PassengerId','Transported']].copy()

In [85]:
df_new.head()

PassengerId  Transported
1209     1288_01            1
4983     5316_01            0
7087     7545_01            1
308      0341_01            1
7740     8260_03            0

In [86]:
len(df_new)

1739

In [87]:
df_new.to_csv('Космический титаник.csv',index=False)